src/main/java/com/mitchej123/hodgepodge/util/StringPooler.java

```diff
@@ -0,0 +1,62 @@
+package com.mitchej123.hodgepodge.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.common.collect.Interner;
+import com.google.common.collect.Interners;
+import com.mitchej123.hodgepodge.config.TweaksConfig;
+
+@SuppressWarnings("UnstableApiUsage")
+public class StringPooler {
+
+    public static StringPooler INSTANCE;
+    private static final Logger LOGGER = LogManager.getLogger("HodgepodgeStringPooler");
+
+    protected StringPooler() {}
+
+    public static void setupPooler() {
+        if (!TweaksConfig.enableStringPooling) {
+            return;
+        }
+        if (INSTANCE != null) {
+            throw new IllegalStateException("Pooler already set up");
+        }
+        LOGGER.info("Setting up string pooler: " + switch (TweaksConfig.stringPoolMode) {
+            case 0 -> " (JVM)";
+            case 1 -> " (Guava strong)";
+            case 2 -> " (Guava weak)";
+            default -> " (unknown)";
+        });
+
+        INSTANCE = switch (TweaksConfig.stringPoolMode) {
+            case 0 -> new StringPooler();
+            case 1 -> new GuavaPooler(true);
+            case 2 -> new GuavaPooler(false);
+            default -> throw new IllegalArgumentException("Invalid mode");
+        };
+    }
+
+    public String getString(String s) {
+        return s.intern();
```

Comment:
Have you actually profiled these? I've been taught string intern is quite a slow process and should only use it in limited scope. Interned string are also kept already forever, essentially creating a memory leak. Surely if they are reused frequently this is not a real memory leak, but I won't just do it to every string value in a NBT tag

Reply:
According to this post using String.intern() this way is a very bad idea: https://stackoverflow.com/questions/10624232/performance-penalty-of-string-intern

Reply:
I did read about this before trying.  The consensus was it became reasonable past Java 7.  Profiling locally showed minimal overhead compared to the memory saved. 

I'd be interested in seeing it on a larger server and compared to the other implementations.

Reply:
Swapped out the default with Guava Strong Pooler; also moved the NBTStringTag pooling to a separate option that is off by default, and only intern NBTTagCompound keys by default as those are far more likely to have duplicates.

Reply:
As a note - even with the NBTStringTag, heavy copy() because of a 22mb Mineshaft.dat, the pooler was a small portion of the time and didn't show up in the sample very often.